include/dlaf/eigensolver/eigensolver/impl.h

```diff
@@ -62,8 +64,19 @@ void Eigensolver<B, D, T>::call(comm::CommunicatorGrid grid, blas::Uplo uplo, Ma
   auto taus = reductionToBand<B>(grid, mat_a, band_size);
   auto ret = bandToTridiag<Backend::MC>(grid, uplo, band_size, mat_a);
 
+#ifdef DLAF_WITH_HDF5
+  matrix::FileHDF5 file(grid.fullCommunicator(), "trid-ref.h5");
+
+  file.write(ret.tridiagonal, "/tridiag");
+#endif
```

Comment:
What about keeping it, and enabling it conditionally on an env variable?
(e.g. DLAF_EIGENSOLVER_WRITE_TRIDIAGONAL)

Reply:
It's not a bad idea, but it might require to slightly change some policy.

Each time a solver runs it will store in the same file, in the same dataset. As it is now implemented, it will error on the second run because in order to write it expects that the file does not exist.

I opted for this behaviour with the aim of not overwriting/truncating existing files involuntarily. But we can change this if we prefer to.

Reply:
> on an env variable

If we do this it should probably also be a configuration option and command line option (perhaps this was implied)?

> not overwriting/truncating existing files involuntarily

What about defaulting the filename to some "unique" filename (with a timestamp or similar)? And perhaps including an option to force overwrite an existing file?

These are extra features though and need not be included in this PR.

Reply:
This is now implemented, right? (In a more elaborate form since you both the environment variable and the command line option.)

Reply:
Right, I'm confusing myself. All good.